sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -0,0 +1,506 @@
+              )
+              .getSql()}) THEN true
+            ELSE false
+        END`,
+        "stopFullTimeBCFunding",
+      )
+      .innerJoin("disbursement.studentAssessment", "studentAssessment")
+      .innerJoin("studentAssessment.application", "application")
+      .innerJoin("application.currentAssessment", "currentAssessment") // * This is to fetch the current assessment of the application, even though we have multiple reassessments
+      .innerJoin("currentAssessment.offering", "offering")
+      .innerJoin("offering.institutionLocation", "institutionLocation")
+      .innerJoin("offering.educationProgram", "educationProgram")
+      .innerJoin("application.student", "student") // ! The student alias here is also used in sub query 'getExistsBlockRestrictionQuery'.
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .innerJoin("application.msfaaNumber", "msfaaNumber")
+      .innerJoin("disbursement.disbursementValues", "disbursementValue")
+      .where("disbursement.dateSent is null")
+      .andWhere("disbursement.disbursementDate <= :disbursementMinDate", {
+        disbursementMinDate,
+      })
+      .andWhere("application.applicationStatus = :applicationStatus", {
+        applicationStatus: ApplicationStatus.completed,
+      })
+      .andWhere("msfaaNumber.dateSigned IS NOT NULL")
+      .andWhere("msfaaNumber.cancelledDate IS NULL")
+      .andWhere("sinValidation.isValidSIN = true")
+      .andWhere("offering.offeringIntensity = :offeringIntensity", {
+        offeringIntensity,
+      })
+      .andWhere(
+        `NOT EXISTS(${this.studentRestrictionService
+          .getExistsBlockRestrictionQuery()
+          .getSql()})`,
+      )
+      .andWhere(
+        "disbursement.disbursementScheduleStatus = :disbursementScheduleStatus",
+        {
+          disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+        },
+      )
+      .andWhere("disbursement.coeStatus = :coeStatus", {
+        coeStatus: COEStatus.completed,
+      })
+      .setParameter("restrictionActionType", stopFullTimeBCFunding)
+      // 'restrictionActions' parameter used inside sub-query.
+      .setParameter("restrictionActions", possibleRestrictionActions)
+      // Necessary, for instance, for the overawards processing.
+      .orderBy("disbursement.disbursementDate")
+      .getRawAndEntities();
+    return mapFromRawAndEntities<ECertDisbursementSchedule>(
+      queryResult,
+      "stopFullTimeBCFunding",
+    );
+  }
+
+  /**
+   * Get the overawards balance consolidated per student and per loan award to
+   * apply, if needed, the overawards deductions.
+   * @param disbursements all disbursements that are part of one e-Cert.
+   * @param entityManager used to execute the commands in the same transaction.
+   */
+  private async applyOverawardsDeductions(
+    disbursements: ECertDisbursementSchedule[],
+    entityManager: EntityManager,
+  ) {
```

Comment:
return type?

Reply:
yes, I saw some PR where using `:void`.  just a suggestion

Reply:
I don't mind adding it if everyone will follow.

Reply:
This one is actually an async method and it was missing the Promise<void>. I just added it.